hydra/client.go

```diff
@@ -207,3 +231,39 @@ func (c *InternalClient) do(req *http.Request, v interface{}) (*http.Response, e
 	}
 	return resp, err
 }
+
+func fetchApiKey(spec hydrav1alpha1.ApiKeySecretRef) (string, error) {
+
+	secretName := spec.Name
+	secretNamespace := spec.Namespace
+
+	if secretName == "" || secretNamespace == "" {
+		return "", nil
+	}
+
+	cfg := ctrl.GetConfigOrDie()
+	kubeClient, err := client.New(cfg, client.Options{})
```

Comment:
Not a fan of this pattern. I think we should define the clients either in the New function, or pass them to the function to avoid the possibility of creating multiple instances. Then we can refactor this function into a method and create some unit tests :)